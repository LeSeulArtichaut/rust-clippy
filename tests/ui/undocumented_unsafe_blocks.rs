#![warn(clippy::undocumented_unsafe_blocks)]
#![allow(unused_unsafe)]

fn good_basic() {
    // SAFETY
    unsafe {}
}

fn good_newlines() {
    // SAFETY
    //
    //
    //
    unsafe {}
}

#[rustfmt::skip]
fn good_whitespaces() {
    // SAFETY



    unsafe {}
}

fn good_block() {
    /* SAFETY */
    unsafe {}
}

#[rustfmt::skip]
fn good_inline_block() {
    /* SAFETY */ unsafe {}
}

fn good_statement() {
    let _ =
        // SAFETY
        unsafe {};
}

fn bad_no_comment() {
    unsafe {}
}

fn bad_comment_after_block() {
    unsafe {} // SAFETY
}

fn bad_comment_after_block2() {
    unsafe {
        //
    } // SAFETY
    // SAFETY
}

fn bad_comment_inside_block() {
    unsafe {
        // SAFETY
    }
}

#[rustfmt::skip]
fn bad_two_blocks_one_comment() {
    // SAFETY
    unsafe {}; unsafe {}
}

#[rustfmt::skip]
fn bad_block_comment_inside_block() {
    unsafe { /* SAFETY */ }
}

fn main() {}
