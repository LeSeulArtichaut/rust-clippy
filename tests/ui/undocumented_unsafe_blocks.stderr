error: this `unsafe` block is missing a SAFETY comment
  --> $DIR/undocumented_unsafe_blocks.rs:43:5
   |
LL |     unsafe {}
   |     ^^^^^^^^^
   |
   = note: `-D clippy::undocumented-unsafe-blocks` implied by `-D warnings`

error: this `unsafe` block is missing a SAFETY comment
  --> $DIR/undocumented_unsafe_blocks.rs:47:5
   |
LL |     unsafe {} // SAFETY
   |     ^^^^^^^^^

error: this `unsafe` block is missing a SAFETY comment
  --> $DIR/undocumented_unsafe_blocks.rs:51:5
   |
LL | /     unsafe {
LL | |         //
LL | |     } // SAFETY
   | |_____^

error: this `unsafe` block is missing a SAFETY comment
  --> $DIR/undocumented_unsafe_blocks.rs:58:5
   |
LL | /     unsafe {
LL | |         // SAFETY
LL | |     }
   | |_____^

error: this `unsafe` block is missing a SAFETY comment
  --> $DIR/undocumented_unsafe_blocks.rs:66:16
   |
LL |     unsafe {}; unsafe {}
   |                ^^^^^^^^^

error: this `unsafe` block is missing a SAFETY comment
  --> $DIR/undocumented_unsafe_blocks.rs:71:5
   |
LL |     unsafe { /* SAFETY */ }
   |     ^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 6 previous errors

